src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -662,6 +662,9 @@
   product(bool, PrintIntrinsics, false, DIAGNOSTIC,                         \
           "prints attempted and successful inlining of intrinsics")         \
                                                                             \
+  develop(bool, VerifyIntrinsicChecks, false,                               \
```

Comment:
We should add testing that uses this new flag. Maybe we could add a run to the tests that check the affected intrinsics? We could also add it to our (Oracle internal) stress test jobs at higher tiers.

Reply:
Given @TobiHartmann's later approval, I'm marking this as resolved.